requirements.txt

```diff
@@ -0,0 +1,5 @@
+setuptools
```

Comment:
AFAIK the only current non-standard packages we use are numpy and matplotlib, not sure if we need setuptools and I don't _think_ we need eth-hash

Reply:
I ran `find . -name '*.py' -exec grep -E '^import [a-zA-Z]+' {} \; | sort | uniq -u` to get a list of python packages in our codebase. Then looked up which were not standard. 
I needed setuptools to get around a distutils being deprecated error for python3.12 (to install pip).